handbook/repository/aws-secrets.md

```diff
@@ -0,0 +1,191 @@
+# AWS Secrets
+
+## Prerequisites
+
+- You will need [AWS command line](https://docs.aws.amazon.com/cli/latest/userguide/install-cliv2.html) installed.
+- You will also need access to the AWS account. Ask someone from DevOps to send you an invitation.
+
+## Getting started
+
+### Using AWS SSO
+
+Using SSO is the most straight forward solution. You won't need to go by yourself on your AWS account and it will open the needed url for you.
+
+1. Run the sso command for the first time
+
+```bash
+aws configure sso
+
+SSO start URL [None]: https://island-is.awsapps.com/start
+SSO Region [None]: eu-west-1
+```
+
+Then choose the environnement of your choice. Likely to be `island-is-development01`. You will be prompted for the following:
+
+```md
+CLI default client Region [eu-west-1]: <Press Enter>
+CLI default output format [json]: <Press Enter>
+CLI profile name [AWSPowerUserAccess-X]: <Custom name (e.g. dev)> or <Press Enter>
+```
+
+This step will add the new profile to your `~/.aws/config` file. If you choose `dev` as profile's name, you will see `[profile dev]` in there.
+
+2. Ready to use
+
+You can now pass your profile to the scripts we have.
+
+```bash
+AWS_PROFILE=<profile-name> yarn get-secrets <project-name> # e.g. profile-name -> dev as seen above
+```
+
+3. Refresh your profile
+
+Once you configured the previous profile, you can run the following command to renew your SSO credentials.
+
+```bash
+aws configure sso --profile <profile-name> # e.g. profile-name -> dev as seen above
+```
+
+It will open the browser, go to your AWS account to log in and will refresh your credentials and you are ready to use the AWS commands again.
+
+{% hint style="warning" %}
+The SSO login expires every 8 hours.
+{% endhint %}
+
+### Using AWS session
+
+This method is more manual where you will need to export environments or change file by yourself.
+
+You will need to go to your [AWS account](https://island-is.awsapps.com/start) and get the required credentials for the account you need.
+
+1. Option 1: Set environments variables
+
+You can first copy/paste theses environments variables on your terminal.
+
+```bash
+export AWS_ACCESS_KEY_ID=AWS_ACCESS_KEY_ID_EXAMPLE
+export AWS_SECRET_ACCESS_KEY=AWS_SECRET_ACCESS_KEY_EXAMPLE
+export AWS_SESSION_TOKEN=AWS_SESSION_TOKEN_EXAMPLE
+```
+
+2. Option 2: Edit `~/.aws/credentials`
+
+Or copy/paste the values in the `~/.aws/credentials` file.
+
+```bash
+[X_AWSPowerUserAccess]
```

Comment:
```suggestion
[default]
```
?

Reply:
Correct, that's a tricky one, AWS website gives a custom profile name instead of default